src/arrays/programs_involving_data_sequences/hard/q8/MainTest.java

```diff
@@ -0,0 +1,96 @@
+import static org.junit.Assert.assertArrayEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomArray(-10, 10, n));
+        return new Clause[]{
+                new StringLiteral("Enter an array of " + n + " integers:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The result of multiplied array is:"),
+                new NewLine(),
+                new PlaceHolder(),
+                new Optional(new StringLiteral(" "))
+        };
+    }
+
+    public void runMain() {
+        MultItUp.main(new String[0]);
+    }
+
+    public static int[] mul(int[] input) {
+        for (int i = 0; i < input.length; i++) {
+            int value = input[i] * i;
+            input[i] = value;
+        }
+        return input;
+    }
+
+    static Stream<Arguments> mainInputProvider() {
+        return Stream.of(
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-1, 2, n)),
+                Arguments.of(replicateArray(1, n)),
+                Arguments.of(replicateArray(0, n)),
+                Arguments.of(replicateArray(10000, n))
+
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainInputProvider")
+    void printCorrectOutput(int[] a) throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct multiplied array.";
+        runWithInput(arrayToInput(a), new Clause[]{
+                new StringLiteral(arrayToInput(mul(a)))
+        });
+    }
+
+    static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(generateRandomArray(-10000, 10000, 10000)),
+                Arguments.of(generateRandomArray(-10000, 10000, 10000)),
+                Arguments.of(generateRandomArray(-10000, 10000, 1000)),
+                Arguments.of(generateRandomArray(-1, 2, 10000)),
+                Arguments.of(replicateArray(1, 10000)),
+                Arguments.of(replicateArray(0, 10000)),
+                Arguments.of(replicateArray(10000, 10000))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    void correctFibonacciMaker(int[] input) throws Throwable {
+        Object[][] arguments = {
+                {input, int[].class},
+        };
+        MethodTest m = new MethodTest(MultItUp.class, "productMaker", arguments);
+        m.setIncorrectMethodStructureErrorMessage("Your productMaker method does not return the correct multiplied array");
+        Object output = m.callMethod();
+        assertArrayEquals((int[]) output, mul(input));
```

Comment:
Currently, there is no way to fail this test. Since the java method alters the array you pass it instead of making a new one, you are grabbing the same reference in the output as is in the input so a comparison between the two will never fail.

Reply:
Also, you should be using CustomAssertions which would not require you to cast output and include an error message in the test.